sky/backends/cloud_vm_ray_backend.py

```diff
@@ -3419,10 +3419,6 @@ def _dump_code_to_file(codegen: str) -> None:
             # already 2x vCPU controller processes running on the controller VM,
```

Comment:
This comment may need to be revamped? It seems not relevant any more? Also, do we still need this pending state setting? It seems our scheduler can do that now?

Reply:
I think this is still useful so that the job appears immediately. There can still be a delay before it is submitted to the scheduler.